src/button/button-group.scss

```diff
@@ -5,72 +5,79 @@
 @mixin iui-button-group {
   display: inline-flex;
   align-items: center;
+  isolation: isolate;
 
-  > .iui-button {
-    &:not(.iui-borderless) {
-      border-radius: 0;
-    }
-
-    &:first-of-type:not(.iui-borderless) {
-      border-top-left-radius: $iui-border-radius;
-      border-bottom-left-radius: $iui-border-radius;
-    }
+  $childSelector: 'input, button'; // target nested inputs and buttons
 
-    &:last-of-type:not(.iui-borderless) {
-      border-top-right-radius: $iui-border-radius;
-      border-bottom-right-radius: $iui-border-radius;
-    }
+  > * {
+    #{$childSelector} {
+      &:hover,
+      &:focus,
+      &:focus-within {
+        position: relative;
+        z-index: 1;
+      }
 
-    &.iui-default:not(:last-of-type) {
-      border-right: none;
-    }
+      &:disabled {
+        position: relative;
+        z-index: -1;
```

Comment:
Is this `z-index` here & above so `:focus` appears above the disabled adjacent element?

Reply:
This is the thought train:

0. negative margins are being used, so by default an element does not have its right border visible
1. focus should be above siblings to show border/focus styles correctly despite negative margin
2. disabled elements should always be below non-disabled siblings regardless of focus, because non-disabled borders should take preference